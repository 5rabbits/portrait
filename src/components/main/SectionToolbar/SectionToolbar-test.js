import { React, expect, mount } from 'test/helper'
import SectionToolbar from 'main/SectionToolbar'

describe('SectionToolbar', () => {
  it('should render with minimum props', () => {
    const component = mount(<SectionToolbar />)

    expect(component).to.be.ok
  })
})
