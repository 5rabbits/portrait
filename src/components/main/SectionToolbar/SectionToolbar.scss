@import '~styles/core';

.SectionToolbar {
  background-color: $body-color;
}

.SectionToolbar__grid {
  height: 80px;
}

.SectionToolbar__actions {
  text-align: right;
}
