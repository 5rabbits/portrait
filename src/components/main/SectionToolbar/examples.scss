.component-example--SectionToolbar {
  [data-reactroot] {
    width: 100%;
  }
}
