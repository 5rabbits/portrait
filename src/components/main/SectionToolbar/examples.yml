description: The most common use case of the section toolbar.
code: >
  <SectionToolbar
    title={<h4>Users</h4>}
    search={<SearchInput placeholder="Write to search" />}
    actions={<Button label="Add user" />}
  />
