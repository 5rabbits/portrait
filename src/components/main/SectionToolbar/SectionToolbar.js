import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Container, Grid } from 'shared'
import './SectionToolbar.scss'

/**
 * This toolbar gives the user immediate access to the most commonly used section actions:
 * searching and creating resources.
 */
export default class SectionToolbar extends PureComponent {
  static propTypes = {
    /**
     * One or more buttons that will be rendered on the right edge of the screen.
     */
    actions: PropTypes.node,

    /**
     * Input primary intented as a full text search field for the section resources.
     */
    search: PropTypes.node,

    /**
     * Section name for context purpose.
     */
    title: PropTypes.node.isRequired,
  }

  static defaultProps = {
    actions: null,
    search: null,
  }

  render() {
    const { actions, search, title } = this.props

    return (
      <Container
        className="SectionToolbar"
        size="fluid"
        >
        <Grid
          className="SectionToolbar__grid"
          align="center"
          size={[1, '0 1 480px', 1]}
          >
          <div className="SectionToolbar__title">{title}</div>
          <div className="SectionToolbar__search">{search}</div>
          <div className="SectionToolbar__actions">{actions}</div>
        </Grid>
      </Container>
    )
  }
}
