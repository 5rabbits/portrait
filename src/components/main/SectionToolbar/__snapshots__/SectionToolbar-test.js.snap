// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionToolbar should render with minimum props 1`] = `
<SectionToolbar
  actions={null}
  search={null}
  title="Test"
>
  <Container
    className="SectionToolbar"
    size="fluid"
  >
    <div
      className="Container--fluid SectionToolbar"
    >
      <Grid
        align="center"
        className="SectionToolbar__grid"
        direction="horizontal"
        gutter="normal"
        justify="start"
        size={
          Array [
            1,
            "0 1 480px",
            1,
          ]
        }
      >
        <div
          className="grid-component SectionToolbar__grid grid-component--center-align grid-component--horizontal-direction grid-component--normal-gutter grid-component--start-justify"
        >
          <div
            className="SectionToolbar__title"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Test
          </div>
          <div
            className="SectionToolbar__search"
            style={
              Object {
                "flex": "0 1 480px",
              }
            }
          />
          <div
            className="SectionToolbar__actions"
            style={
              Object {
                "flex": 1,
              }
            }
          />
        </div>
      </Grid>
    </div>
  </Container>
</SectionToolbar>
`;
